clientapi/routing/register.go

```diff
@@ -225,17 +224,49 @@ func validateApplicationServiceUsername(username string) *util.JSONResponse {
 }
 
 // validatePassword returns an error response if the password is invalid
-func validatePassword(password string) *util.JSONResponse {
+func validatePassword(password string, cfg config.PasswordRequirements) *util.JSONResponse {
 	// https://github.com/matrix-org/synapse/blob/v0.20.0/synapse/rest/client/v2_alpha/register.py#L161
-	if len(password) > maxPasswordLength {
+
+	if len(password) > cfg.MaxPasswordLength {
+		return &util.JSONResponse{
+			Code: http.StatusBadRequest,
+			JSON: jsonerror.WeakPassword(fmt.Sprintf("'password' >%d characters", cfg.MaxPasswordLength)),
+		}
+	} else if len(password) >= 0 && len(password) < cfg.MinPasswordLength {
+		return &util.JSONResponse{
+			Code: http.StatusBadRequest,
+			JSON: jsonerror.WeakPassword(fmt.Sprintf("password too weak: min %d chars", cfg.MinPasswordLength)),
+		}
+	}
+
+	var (
+		symbolCount    = 0
+		uppercaseCount = 0
+		lowercaseCount = 0
+	)
+
+	for _, letter := range password {
+		switch {
+		case unicode.IsSymbol(letter) || unicode.IsPunct(letter):
```

Comment:
@kegsay -- ended up changing this to use the `unicode` package instead, as I feel like its less anglocentric here. would love to get your thoughts.

Reply:
Seems reasonable.